unified_planning/model/problem.py

```diff
@@ -558,20 +569,67 @@ def add_goal(
         if goal_exp != self._env.expression_manager.TRUE():
             self._goals.append(goal_exp)
 
+    def add_trajectory_constraint(self, constraint: "up.model.fnode.FNode"):
```

Comment:
```suggestion
    def add_trajectory_constraint(self, constraint: "up.model.expression.BoolExpression"):
```
Since you call the `auto_promote` method, you can accept every BoolExpression, even if it is not already an `FNode`